File: ruby/src/pizza.rb
class Pizza
    attr_accessor :layers, :baked

    def initialize
        @layers = []
    end
                                                         
    def spread(ingredient)
        @layers << ingredient
    end

    def toppings(*ingredients)
        @layers << ingredients
    end

    def baked
        @baked = true
    end

    def method_missing(name) 
        name.to_s
    end
    
    def to_s
        "The layers of this pizza are #{layers.join(", ")}"
    end
end

File: scala/src/main/scala/pizza/Pizza.scala
package com.pizza

import scala.collection.mutable.ListBuffer

class Pizza {
   var layers = new ListBuffer[String]()
   var baked = false

   def spread(ingredient : String) = {
       layers += ingredient
       this
   }

   def toppings(ingredients: String*) = {
       layers ++= ingredients.to[ListBuffer]
       this
   }

   def bake() = {
       baked = true
       this
   }

   override def toString() : String = {
     "The layers on this pizza are: " + layers.mkString(", ")
   }

}
